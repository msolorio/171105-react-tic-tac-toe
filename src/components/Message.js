import React from 'react';
import PropTypes from 'prop-types';

export default function Message(props) {

  function getMessage() {
    switch(true) {
      // converts to a boolean value
      case !!props.winner:
        return `Winner: ${props.winner}`;
      case !props.squaresAvailable:
        return `Game over. There are no available squares.`;
      default:
        return `Player's turn: ${props.xIsNext ? 'X' : 'O'}`;
    }
  }

  return (
    <div className="Message">{getMessage()}</div>
  );
}

Message.propTypes = {
  winner: PropTypes.string.isRequired,
  squaresAvailable: PropTypes.bool.isRequired,
  xIsNext: PropTypes.bool.isRequired
};
