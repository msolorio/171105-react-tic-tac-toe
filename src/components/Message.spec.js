import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Message from './Message';


describe('Message', () => {
  it('should match snapshot', () => {
    const messageProps = {
      winner: '',
      squaresAvailable: true,
      xIsNext: true
    };
    const tree = renderer.create(<Message {...messageProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should display the correct player\'s turn during gameplay', () => {

    const testCases = [
      {
        xIsNext: true,
        text: 'Player\'s turn: X'
      },
      {
        xIsNext: false,
        text: 'Player\'s turn: O'
      }
    ];

    testCases.forEach((testCase) => {
      const messageProps = {
        winner: '',
        squaresAvailable: true,
        xIsNext: testCase.xIsNext
      };
      const message = shallow(<Message {...messageProps} />);

      expect(message.text()).toBe(testCase.text);
    });
  });

  it('should display the correct message if there is a winner', () => {
    const testCases = [
      {
        winner: 'X',
        xIsNext: false,
        text: 'Winner: X'
      },
      {
        winner: 'O',
        xIsNext: false,
        text: 'Winner: O'
      },
      {
        winner: 'X',
        xIsNext: true,
        text: 'Winner: X'
      },
      {
        winner: 'O',
        xIsNext: true,
        text: 'Winner: O'
      }
    ];

    testCases.forEach((testCase) => {
      const messageProps = {
        winner: testCase.winner,
        squaresAvailable: true,
        xIsNext: testCase.xIsNext
      };
      const message = shallow(<Message {...messageProps} />);

      expect(message.text()).toBe(testCase.text);
    });
  });

  it('should display the correct message if there are no squares available', () => {
    const messageProps = {
      winner: '',
      xIsNext: true,
      squaresAvailable: false
    };

    const message = shallow(<Message {...messageProps} />);

    expect(message.text()).toBe('Game over. There are no available squares.')
  });
});
