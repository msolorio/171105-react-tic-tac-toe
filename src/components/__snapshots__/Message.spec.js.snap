// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match snapshot 1`] = `
<div
  className="Message"
>
  Player's turn: X
</div>
`;
